Model/Checkout/PayOrder.php

```diff
@@ -174,12 +156,12 @@ public function execute(int $orderId, string $amwalOrderId) : bool
             ->setLastRealOrderId($order->getIncrementId())
             ->setLastOrderStatus($order->getStatus());
 
-        $this->orderRepository->save($order);
+        $this->orderUpdate->update($order, $amwalOrderStatus, $historyComment);
+
         if($amwalOrderStatus == 'success') {
-            $this->invoiceAmwalOrder->execute($order, $amwalOrderData);
-            $this->setOrderUrl($order, $amwalOrderId);
             $quote->removeAllItems();
             $this->quoteRepository->save($quote);
+            $this->setOrderUrl($order, $amwalOrderId);
```

Comment:
this needs to move from here to `OrderUpdate` so that all places that call OrderUpdate also get the OrderUrl updated in our backend